tidy3d/components/grid/mesher.py

```diff
@@ -673,7 +673,9 @@ def filter_min_step(
         coords_filter = [interval_coords[0]]
         steps_filter = []
         for coord_ind, coord in enumerate(interval_coords[1:]):
-            if coord - coords_filter[-1] >= min_step:
+            if coord - coords_filter[-1] >= min_step or (
+                coord_ind == len(interval_coords) - 2 and not isclose(coord, coords_filter[-1])
+            ):
```

Comment:
@momchil-flex i lost track here on why we are applying this method to filter small steps. It looks like handled already in `insert_bbox`?

Reply:
I guess at the very least there is this? https://github.com/flexcompute/tidy3d/blob/be7ededc7554a01e3319efa074c03a5107f735d2/tidy3d/components/grid/mesher.py#L669-L670

I do not remember either right now if this is the only reason or there was another use case. But it seems like it should be done even just because of this?

Reply:
i see, that makes sense